test/functional/http/post/5.dapps.js

```diff
@@ -0,0 +1,503 @@
+	var badTransactionsEnforcement = [];
+	var goodTransactionsEnforcement = [];
+
+	var account = node.randomAccount();
+	var accountNoFunds = node.randomAccount();
+	var accountMinimalFunds = node.randomAccount();
+
+	var dappDuplicate = node.randomApplication();
+	var dappDuplicateNameSuccess = node.randomApplication();
+	var dappDuplicateNameFail = node.randomApplication();
+	dappDuplicateNameSuccess.name = dappDuplicateNameFail.name;
+	var dappDuplicateLinkSuccess = node.randomApplication();
+	var dappDuplicateLinkFail = node.randomApplication();
+	dappDuplicateLinkSuccess.link = dappDuplicateLinkFail.link;
+
+	// Crediting accounts
+	before(function () {
+		var promises = [];
+		promises.push(creditAccountPromise(account.address, 1000 * node.normalizer));
+		promises.push(creditAccountPromise(accountMinimalFunds.address, constants.fees.dappRegistration));
+
+		return node.Promise.all(promises)
+			.then(function (results) {
+				results.forEach(function (res) {
+					node.expect(res).to.have.property('success').to.be.ok;
+					node.expect(res).to.have.property('transactionId').that.is.not.empty;
+					transactionsToWaitFor.push(res.transactionId);
+				});
+			})
+			.then(function (res) {
+				return waitForConfirmations(transactionsToWaitFor);
+			});
+	});
+
+	describe('schema validations', function () {
+
+		shared.invalidAssets(account, 'dapp', badTransactions);
+
+		describe('category', function () {
+			
+			it('without should fail', function () {
+				transaction = node.lisk.dapp.createDapp(account.password, null, node.randomApplication());
+				delete transaction.asset.dapp.category;
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('success').to.be.not.ok;
+					node.expect(res).to.have.property('message').to.match(/Missing required property: category$/);
+					badTransactions.push(transaction);
+				});
+			});
+
+			it('with integer less than minimum should fail', function () {
+				transaction = node.lisk.dapp.createDapp(account.password, null, node.randomApplication());
+				transaction.asset.dapp.category = -1;
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('success').to.be.not.ok;
+					node.expect(res).to.have.property('message').to.match(/Value -1 is less than minimum 0$/);
+					badTransactions.push(transaction);
+				});
+			});
+
+			it('with integer greater than maximum should fail', function () {
+				transaction = node.lisk.dapp.createDapp(account.password, null, node.randomApplication());
+				transaction.asset.dapp.category = 9;
+
+				return sendTransactionPromise(transaction).then(function (res) {
+					node.expect(res).to.have.property('success').to.be.not.ok;
+					node.expect(res).to.have.property('message').to.match(/Value 9 is greater than maximum 8$/);
+					badTransactions.push(transaction);
+				});
+			});
+		});
+
+		describe('description', function () {
```

Comment:
Missing test for a non-string description

Reply:
Lisk-js does not allow an integer.

Reply:
Overwrite the field then after a transaction is created using lisk-js